### PR TITLE
TES-453 Added configuration of Application Insights

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description

Added configuration of Application Insights

## Related Issues

TES-453

## Changes

Added configuration for Application Insights, alerts, action groups and smart detection rules

## Screenshots (if applicable)

N/A

## Checklist

- [x] I have tested these changes thoroughly.
- [x] My code follows the project's coding style.
- [x] I have added appropriate comments to my code, especially in complex areas.
- [x] All new and existing tests passed locally.

### CODE REVIEW

File: modules/monitoring/output.tf [hunk 1]
```diff
@@ -2,3 +2,7 @@ output "log_analytics_workspace_id" {
   description = "Outputs log analytics workspace identifier"
   value       = azurerm_log_analytics_workspace.log_analytics_workspace.id
 }
+
+output "connection_string" {
```
Comment (mihailradkov) [inline modules/monitoring/output.tf:6]: `connection_string` -> `appi_connection_string` + description

File: modules/user-data/main.tf [hunk 2]
```diff
@@ -48,16 +52,26 @@ data "cloudinit_config" "entrypoint" {
...
+  # 06 Application Insights configuration
   part {
     content_type = "text/x-shellscript"
-    content = templatefile("${path.module}/templates/06_cluster_setup.sh.tpl", {
+    content = templatefile("${path.module}/templates/06_application_insights_config.sh.tpl", {
+      appi_connection_string : var.appi_connection_string
+    })
+  }
+
+  # 02 Cluster setup
```
Comment (mihailradkov) [inline modules/user-data/main.tf:71]: 7

File: variables.tf [hunk 3]
```diff
@@ -293,3 +293,9 @@ variable "disk_public_network_access" {
   type        = bool
   default     = false
 }
+
+variable "alerts_email_recipients" {
+  description = "List of e-mail recipients for alerts"
+  type = list(string)
+  default = ["tes@ontotext.com"]
```
Comment (mihailradkov) [inline variables.tf:300]: Noooo :laughing:  We don't want the spam!  Leave it an empty `[]`

File: modules/monitoring/main.tf [hunk 4]
```diff
@@ -12,179 +12,99 @@ resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
...
-    }
+resource "azurerm_application_insights_smart_detection_rule" "rise_daily_data_volume" {
+  name                        = "Abnormal rise in daily data volume"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
+}
 
-    log_file {
-      name          = local.gdb_main_log_table_name
```
Comment (mihailradkov) [inline modules/monitoring/main.tf:65]: gdb_main_log_table_name is now a dangling local

File: modules/monitoring/main.tf [hunk 5]
```diff
@@ -12,179 +12,99 @@ resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
...
 
-resource "azurerm_monitor_data_collection_rule_association" "DCR_association" {
-  name                    = "dcra-${var.resource_group_name}"
-  target_resource_id      = var.vmss_resource_id
-  data_collection_rule_id = azurerm_monitor_data_collection_rule.data_collection_rule.id
-  description             = "Associates DCR to the VMSS"
-}
 
-# We need to use the azapi as terraform does not have a resource to create tables.
-resource "azapi_resource" "main_gdb_log_table" {
```
Comment (mihailradkov) [inline modules/monitoring/main.tf:161]: You no longer need the provider for azapi

File: modules/user-data/main.tf (hunk 2 above)
Comment (viktor-ribchev) [reply modules/user-data/main.tf]: yep

File: variables.tf (hunk 3 above)
Comment (viktor-ribchev) [reply variables.tf]: sure

File: modules/monitoring/main.tf (hunk 4 above)
Comment (viktor-ribchev) [reply modules/monitoring/main.tf:65]: removed

File: modules/monitoring/main.tf (hunk 5 above)
Comment (viktor-ribchev) [reply modules/monitoring/main.tf:161]: removed

File: modules/monitoring/output.tf [hunk 6]
```diff
@@ -2,3 +2,7 @@ output "log_analytics_workspace_id" {
   description = "Outputs log analytics workspace identifier"
   value       = azurerm_log_analytics_workspace.log_analytics_workspace.id
 }
+
+output "appi_connection_string" {
+  value = azurerm_application_insights.graphdb_insights.connection_string
```
Comment (mihailradkov) [inline modules/monitoring/output.tf:7]: add description

File: variables.tf [hunk 7]
```diff
@@ -293,3 +293,21 @@ variable "disk_public_network_access" {
   type        = bool
   default     = false
 }
+
+variable "alerts_email_recipients" {
+  description = "List of e-mail recipients for alerts"
+  type        = list(string)
+  default     = ["tes@ontotext.com"]
```
Comment (mihailradkov) [inline variables.tf:300]: image

**Review by mihailradkov — CHANGES_REQUESTED** [review]
Address the remaining comments

File: modules/monitoring/output.tf (hunk 6 above)
Comment (viktor-ribchev) [reply modules/monitoring/output.tf]: added

File: variables.tf (hunk 7 above)
Comment (viktor-ribchev) [reply variables.tf]: removed

**Review by mihailradkov — APPROVED** [review]

**Review by yaskoo — APPROVED** [review]

**Review by yaskoo — APPROVED** [review]

**Pull request merged**